contacts_school_permission/models/res_partner_permission.py

```diff
@@ -48,6 +61,19 @@ def _compute_allowed_signer_ids(self):
                     lambda l: l.relation in ('progenitor', 'guardian')
                 ).mapped('responsible_id'))
 
+    @api.depends('signer_id', 'signer_id_2')
+    def _compute_signer_ids(self):
+        for record in self:
+            if self.signature:
+                record.signer_ids |= self.signer_id
+            if self.signature_2:
+                record.signer_ids |= self.signer_id_2
```

Comment:
Esto no es correcto

* si los campos dependen de los _signatures_ tienes que ponerlos en el depends
* en vez de self, tienes que poner record, ya que estas dentro del **for**